services/wireguard/service/service_unix.go

```diff
@@ -267,36 +266,45 @@ func (m *Manager) addTraversalParams(config wg.ServiceConfig, traversalParams tr
 	// There is no need to add any connect delay when port mapping failed.
 	config.Consumer.ConnectDelay = 0
 
+	// TODO this backward compatibility block needs to be removed once we will start using port ranges for all peers.
+	config.LocalPort = traversalParams.RemotePorts[len(traversalParams.RemotePorts)-1]
```

Comment:
Are you sure you wont index out of range exception here?

Reply:
Added a check just in case.